graphql_compiler/tests/test_ir_generation_errors.py

```diff
@@ -1034,3 +1036,35 @@ def test_invalid_edge_degree_queries(self):
         for invalid_graphql in invalid_queries:
             with self.assertRaises(GraphQLCompilationError):
                 graphql_to_ir(self.schema, invalid_graphql)
+
+    def test_directives_not_in_schema(self):
```

Comment:
I'd also add maybe three more tests:
- unexpected directive that the compiler doesn't know what to do with
- recognized directive but with wrong "locations" (FIELD vs INLINE_FRAGMENT vs both)
- recognized directive but with too many / too few / wrong arguments